coalib/core/PersistentHash.py

```diff
@@ -1,7 +1,53 @@
+from collections import Iterable
 from hashlib import sha1
 import pickle
 
 
+def tuple_replace(tup, x, y):
+    tup_list = list(tup)
+    for element in tup_list:
+        if element == x:
+            tup_list[tup_list.index(element)] = y
+    new_tuple = tuple(tup_list)
+    return new_tuple
+
+
+def preserve_order(obj):
+    args_tuple = obj[0]
+    sorted_args = args_tuple
+
+    # iterates over the tuple of positional arguments
+    # looks for sets and dicts, sort them and put them
+    # back into the tuple.
+    for element in sorted_args:
+        if isinstance(element, Iterable):
+            if any(element) and isinstance(element, dict):
+                sorted_args = tuple_replace(sorted_args, element,
+                                            dict(sorted(element.items())))
+            if any(element) and isinstance(element, set):
+                sorted_args = tuple_replace(sorted_args, element,
+                                            sorted(element))
+
+    obj = tuple_replace(obj, args_tuple, sorted_args)
+
+    kwargs_dict = obj[1]
+    sorted_kwargs = kwargs_dict
+    # iterates over the keyword arguments dict
+    # looks for sets and dicts within the kwargs dict
+    # sort each individual set or dict and puts them
+    # back them into the dict.
+    for key, value in sorted_kwargs.items():
+        if any(value) and isinstance(value, dict):
+            sorted_kwargs[key] = dict(sorted(value.items()))
+        if any(value) and isinstance(value, set):
+            sorted_kwargs[key] = sorted(value)
+
+    sorted_kwargs = dict(sorted(sorted_kwargs.items()))
```

Comment:
again you make the sorting obsolete again by converting the stuff back into a dict.

Reply:
Probably that's the cause why you receive different results on different computers in your test, I guess this is the crucial part here (and the other similar statement above)